Content.Client/Graphics/Overlays/CircleMaskOverlay.cs

```diff
@@ -13,18 +14,21 @@ public class CircleMaskOverlay : Overlay
     {
         [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
         [Dependency] private readonly IEyeManager _eyeManager = default!;
+        [Dependency] private readonly IPlayerManager _playerManager = default!;
 
         public override OverlaySpace Space => OverlaySpace.WorldSpace;
         private readonly ShaderInstance _shader;
 
-        public CircleMaskOverlay() : base(nameof(SharedOverlayID.CircleMaskOverlay))
+        public CircleMaskOverlay() : base(nameof(CircleMaskOverlay))
         {
             IoCManager.InjectDependencies(this);
             _shader = _prototypeManager.Index<ShaderPrototype>("CircleMask").Instance();
         }
 
         protected override void Draw(DrawingHandleBase handle, OverlaySpace currentSpace)
         {
+            if (!GradientCircleMaskOverlay.LocalPlayerHasState(_playerManager, false, true))
+                return;
```

Comment:
This feels hacky.

Reply:
see other responses. Having everything be stateful caused bugs when the inevitable out-of-sync-ness happened. This is why ghosts end up with masks still active.

Reply:
Isn't this masking the problem then though if they're entering a state multiple times?

Reply:
They're not entering a state multiple times.
What happens is that:
+ inevitably when something happens that affects the overlay effects manager (like the player switching eyes/bodies) the overlays get carried over because they aren't properly shutdown.
+ People put in stuff to remove overlays in these specific cases as a workaround and it obviously didn't work.
+ *The most important: The states kept completely clearing all overlays on remove, so it wasn't possible to use overlay effects component safely for anything other than states anyway.*